src/udx.c

```diff
@@ -841,6 +882,11 @@ on_uv_poll (uv_poll_t *handle, int status, int events) {
 
     if (adjust_ttl) uv_udp_set_ttl((uv_udp_t *) socket, pkt->ttl);
 
+    if (socket->family == 6 && pkt->dest.ss_family == AF_INET) {
+      addr_to_v6((struct sockaddr_in *) &(pkt->dest));
+      pkt->dest_len = sizeof(struct sockaddr_in6);
+    }
```

Comment:
does every packet sent to an ipv4 addr go through this if dualstack? and if so shouldn't we make the stream dest an ipv6 mapped dest then on connect?

Reply:
Good point, that'd be a good optimisation!